prov/verbs/src/ep_rdm/verbs_rdm_cm.c

```diff
@@ -646,10 +646,20 @@ fi_ibv_rdm_process_event_rejected(struct fi_ibv_rdm_ep *ep,
 {
 	struct fi_ibv_rdm_tagged_conn *conn = event->id->context;
 	ssize_t ret = FI_SUCCESS;
+	const int *pdata = NULL;
+
+	if (event->param.conn.private_data) {
```

Comment:
Do you mean keep condition as before and add new parts with workaround? Sorry, but I disagree - this looks awful (at least for me). This code works only for connection establishing (not a critical path) and extra condition don't have any significant overhead here. So, I would prefer to have clear extra code.